### PR TITLE
feat(ui/message): custom markdown -> html parser

### PR DESCRIPTION
**Pull request by JoltCode**
# Description

Converts markdown to html

Fixes N/A (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

TBC

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have updated .env.example if I added a new environment variable
- [x] My PR title follows the [semantic commits](https://www.conventionalcommits.org/en/v1.0.0/) style

### CODE REVIEW

File: global.css [hunk 1]
```diff
@@ -68,13 +68,7 @@
...
 
-.hljs {
-	@apply my-4 w-full rounded-lg bg-neutral-100 text-sm dark:bg-neutral-900 !important;
-	/* Fixes overflowing code blocks */
-	white-space: break-spaces;
-	line-break: anywhere;
-	font-family: 'Courier New', Courier, monospace;
-}
+
 .discord-message a {
```
Comment (RhysSullivan) [inline global.css:72]: Probs should be removed and moved into the parser

File: global.css [hunk 2]
```diff
@@ -84,12 +78,12 @@
 .d-emoji {
```
Comment (RhysSullivan) [inline global.css:78]: Move into the parser

File: global.css [hunk 3]
```diff
@@ -84,12 +78,12 @@
...
 	@apply inline-block h-6 w-6;
 }
-.hljs-built_in{
-  @apply text-red-700 dark:text-red-400 !important;
-}
-.hljs-keyword{
-  @apply text-red-700 dark:text-red-400 !important;
-}
+/*.hljs-built_in{*/
```
Comment (RhysSullivan) [inline global.css:81]: Delete

File: packages/ui/.ladle/components.tsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import type { ArgTypes, GlobalProvider } from '@ladle/react';
 import {
 	WithAnalytics,
 	WithAuth,
-	WithHighlightJS,
+	WIthMarkdown,
```
Comment (RhysSullivan) [inline packages/ui/.ladle/components.tsx:7]: Typo should be WithMarkdown

File: packages/ui/src/components/pages/MessageResultPage.tsx [hunk 5]
```diff
@@ -18,11 +18,15 @@ import {
 import { isServer } from '~ui/utils/checks';
 import Head from 'next/head';
 import type { QAPage, WithContext } from 'schema-dts';
+<<<<<<< HEAD
```
Comment (RhysSullivan) [inline packages/ui/src/components/pages/MessageResultPage.tsx:21]: Missed these git merge markers

File: packages/ui/src/components/pages/MessageResultPage.tsx [hunk 6]
```diff
@@ -170,19 +175,24 @@ export function MessageResultPage({
...
+	const solutionHtml = useParsedMarkdown(solution?.content ?? '');
+
 	const qaHeader: WithContext<QAPage> = {
 		'@context': 'https://schema.org',
 		'@type': 'QAPage',
 		mainEntity: {
 			'@type': 'Question',
-			name: toHTML(question),
-			text: toHTML(firstMessage.content),
+			name: nameHtml,
```
Comment (RhysSullivan) [inline packages/ui/src/components/pages/MessageResultPage.tsx:188]: I have some concerns about moving the sanitization away from where it's being used - however - maybe it'd be ok if we make this var name  like "safeNameHtml"

**Review by RhysSullivan — CHANGES_REQUESTED** [review]
First pass changes

File: packages/ui/src/components/pages/MessageResultPage.tsx (hunk 5 above)
Comment (JoltCode) [reply packages/ui/src/components/pages/MessageResultPage.tsx:21]: Oops 😅

File: packages/ui/.ladle/components.tsx (hunk 4 above)
Comment (JoltCode) [reply packages/ui/.ladle/components.tsx:7]: Ahhh haha will fix 👍

File: global.css (hunk 1 above)
Comment (JoltCode) [reply global.css:72]: Will do 👍

**Comment by JoltCode** [comment]
Another review please :smile:

**Comment by RhysSullivan** [comment]
image
Needs light theme support

**Comment by RhysSullivan** [comment]
image
underline doesnt work

**Comment by RhysSullivan** [comment]
image
title doesnt work

**Comment by RhysSullivan** [comment]
image
quote doesnt work

**Comment by JoltCode** [comment]
> image Needs light theme support

We discussed this before and you said this wasn't a problem - for v1 of this do we need light theme support?

**Comment by JoltCode** [comment]
And yes I will fix the rest :+1:

**Comment by JoltCode** [comment]
> image underline doesnt work

Fixed :+1:

**Comment by JoltCode** [comment]
> image title doesnt work

Fixed :+1:

**Comment by JoltCode** [comment]
> image quote doesnt work

Fixed :+1:

**Pull request closed without merging**